src/lib/auth/oauth-client.mjs

```diff
@@ -127,11 +115,11 @@ class OAuthClient {
 
   async start() {
     try {
-      this.server.on("listening", () => {
-        this.port = this.server.address().port;
-        this.server.emit("ready");
-      });
       if (!this.server.listening) {
+        this.server.on("listening", () => {
+          this.port = this.server.address().port;
+          this.server.emit("ready");
+        });
         this.server.listen(0);
```

Comment:
is `this.server.listening` set when you emit `ready`?

Reply:
this check was a workaround for that builtYargs.argv weirdness, so not strictly necessary but yes by the time i send "ready" it's already `true`